### PR TITLE
x/evm: StateBD tests

### PR DESCRIPTION
**Pull request by fedekunze**
Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/types/statedb_test.go [hunk 1]
```diff
@@ -614,3 +615,51 @@ func (suite *StateDBTestSuite) TestCommitStateDB_Finalize() {
...
+		{
+			"aggregate state",
+			suite.address,
+			func() {
+				for i := 0; i < 5; i++ {
+					key := ethcmn.BytesToHash([]byte(fmt.Sprintf("key%d", i)))
+					value := ethcmn.BytesToHash([]byte(fmt.Sprintf("value%d", i)))
+					suite.stateDB.SetState(suite.address, key, value)
+				}
+				suite.stateDB.Finalise(false) // commit state
```
Comment (fedekunze) [inline x/evm/types/statedb_test.go]: this panics

File: x/evm/types/statedb.go [hunk 2]
```diff
@@ -745,13 +744,8 @@ func (csdb *CommitStateDB) ForEachStorage(addr ethcmn.Address, cb func(key, valu
 			continue
 		}
 
-		_, content, _, err := rlp.Split(value.Bytes())
-		if err != nil {
-			return err
-		}
```
Comment (fedekunze) [inline x/evm/types/statedb.go:751]: idk why I added this originally (prob to match geth), the value bytes are not RLP encoded. This should fix the issue with the contract deployments

File: x/evm/types/state_object.go [hunk 3]
```diff
@@ -142,6 +142,9 @@ func (so *stateObject) setState(key, value ethcmn.Hash) {
 	so.dirtyStorage = append(so.dirtyStorage, NewState(key, value))
 	idx = len(so.dirtyStorage) - 1
 	so.keyToDirtyStorageIndex[key] = idx
+	idx = len(so.originStorage)
```
Comment (fedekunze) [inline x/evm/types/state_object.go]: @noot should this be:
```suggestion
	idx = len(so.originStorage) - 1
```

File: x/evm/types/state_object.go [hunk 4]
```diff
@@ -268,6 +270,8 @@ func (so *stateObject) commitState() {
 
 		store.Set(state.Key.Bytes(), state.Value.Bytes())
 	}
+	// clen storage as all entries are dirty
```
Comment (fedekunze) [inline x/evm/types/state_object.go]:
```suggestion
	// clean storage as all entries are dirty
```

File: x/evm/types/state_object.go (hunk 3 above)
Comment (noot) [reply x/evm/types/state_object.go]: moved `append` line above this, then changed to `idx = len(so.originStorage) - 1`

**Comment by fedekunze** [comment]
@noot lmk if this works ok for you

**Review by noot — APPROVED** [review]
works for me!

**Pull request merged**